pkg/prowgen/prowgen_tests.go

```diff
@@ -48,27 +52,29 @@ func DiscoverTests(r Repository, openShift OpenShift, sourceImageName string, sk
 				testTimeout = &prowapi.Duration{Duration: 4 * time.Hour}
 			}
 
-			clusterClaim := &cioperatorapi.ClusterClaim{
-				Product:      cioperatorapi.ReleaseProductOCP,
-				Version:      openShift.Version,
-				Architecture: cioperatorapi.ReleaseArchitectureAMD64,
-				Cloud:        cioperatorapi.CloudAWS,
-				Owner:        "openshift-ci",
-				Timeout:      &prowapi.Duration{Duration: time.Hour},
-			}
-			var clusterProfile cioperatorapi.ClusterProfile
-			workflow := pointer.String("generic-claim")
-			var env cioperatorapi.TestEnvironment
-			if openShift.CandidateRelease {
-				clusterClaim = nil
-				if strings.Contains(r.RepositoryDirectory(), "serverless-operator") {
-					// Use the shared profile for s-o.
-					clusterProfile = "aws"
-				} else {
-					clusterProfile = "aws-serverless"
-					env = map[string]string{
-						"BASE_DOMAIN": devclusterBaseDomain,
-					}
+			var (
+				clusterClaim   *cioperatorapi.ClusterClaim
+				clusterProfile cioperatorapi.ClusterProfile
+				workflow       *string
+				env            cioperatorapi.TestEnvironment
+			)
+
+			useClusterPool := openShift.Version == clusterPoolVersion
+			// Make sure to use the existing cluster pool if available for the given OpenShift version.
+			if useClusterPool {
+				clusterClaim = &cioperatorapi.ClusterClaim{
+					Product:      cioperatorapi.ReleaseProductOCP,
+					Version:      openShift.Version,
+					Architecture: cioperatorapi.ReleaseArchitectureAMD64,
+					Cloud:        cioperatorapi.CloudAWS,
+					Owner:        "serverless-ci",
```

Comment:
might be worth to add a comment here that links to the place in openshift/release that defines `serverless-ci`. Just for later people to understand the link.

Reply:
Yeah, so I will update this PR after https://github.com/openshift/release/pull/49904 is merged so that I can directly point to where it's defined in the codebase.